### DISCUSSION TITLE
Pr3sence Web3 ad network integration

### DISCUSSION THREAD

**Discussion by NBayle1** [rainbow-me/rainbowkit#1568] [General]
Hey RainbowKit team,

Im co-fonder of Pr3sence, we are a web3 ad network that enables super powerful on-chain targeting of ads to users. We host ads directly on dapps, so that when a user connects their wallet, the ad can be personalised to the user based on their on-chain data. E.g. show an ad to specific users that hold X token above X balance, or hold a certain NFT or interacted with a specific project, etc.

We are looking for publishers to host our ads, and exploring potential integration with wallet-connected interfaces across the web3 ecosystem. we would like explore the potential for integrating our ad support into the rainbowkit. This could enable a source of revenue for rainbow kit themselves, and or the project integrating it. Furthermore, it could be used as a form of captcha, for example show an ad, then ask the user what was the last ad you saw, to filter out bots.

We'd love to speak further and explore with the team. Thanks in advance.
- Nico from the Pr3sence team.

**Comment by DanielSinclair** [comment]
@NBayle1 Thanks for reaching out. This is unfortunately not something we are interested in first party support for at this time. But RainbowKit works alongside other sister packages atop of Wagmi, so you can publish an integration, and dApps can adopt your product independently.
